drivers/net/loopback.c

```diff
@@ -69,7 +69,7 @@ static int loopback_send(struct net_if *iface, struct net_pkt *pkt)
 	 * must be dropped. This is very much needed for TCP packets where
 	 * the packet is reference counted in various stages of sending.
 	 */
-	cloned = net_pkt_clone(pkt, K_MSEC(100));
+	cloned = net_pkt_clone(pkt, K_MSEC(100), NET_PKT_CLONE_USER_DATA);
```

Comment:
This is the pattern we are forced to use with github. In order to test the feature, all relevant patches need to be inside one PR (even if they are not fully related). This makes reviewing a bit painful but slightly faster to merge. I am personally trying to get gPTP patches reviewed but it is very slow because I am trying to send patches in smaller chunks which serializes the whole process.

Reply:
> This is the pattern we are forced to use with github.

No, why suddenly? We used to work with Gerrit, which required a review per commit, and everything was "fine".

 > In order to test the feature, all relevant patches need to be inside one PR (even if they are not fully related).

I'm not sure what you mean by "test". If you effectively mean "try/review", then splitting to another PR only makes it easier. And if you mean "test" literally, then we have unittest to check for regressions (and ideally, new tests should be added for new functionality, which we're behind of doing, but still easier to do in a small focused PR).

> This makes reviewing a bit painful but slightly faster to merge.

It's hardly possible to review such PRs thoroughly. How it worked before is:

* Not making thorough reviews
* Ignoring comments and concerns raised

So no, it doesn't make merging faster, but definitely makes it lower quality.

> I am personally trying to get gPTP patches reviewed but it is very slow because I am trying to send patches in smaller chunks which serializes the whole process.

I don't think it's related. It's because of overall complexity of the matter, and that not everyone is familiar with the requirements for this protocol implementation in Zephyr, or with protocol itself. For example, I feel a bit shy to give +1 for it. (But then I'm sure there's no -1 at least in the initial patch I looked it, thanks to it being contained and easy to review).